pkg/controlplane/authz/manager.go

```diff
@@ -173,7 +173,7 @@ func (m *Manager) AddAccessPolicy(policy *api.Policy) error {
 // DeleteAccessPolicy removes an access policy to allow/deny specific connections.
 // TODO: switch from api.Policy to v1alpha1.Policy.
 func (m *Manager) DeleteAccessPolicy(policy *api.Policy) error {
-	return m.policyDecider.DeleteAccessPolicy(policy)
+	return m.policyDecider.DeleteAccessPolicy(policy.Name)
```

Comment:
Unsure if the name alone is sufficient unless it also encodes the namespace.
Even if it does, we can have the same name in a single namespace (privileged and normal are different CRDs).
Seems safer to pass along the entire object (CRD instance).

Unlike the original policy management code, which only allowed a single entry of a policy name - privileged or normal, with CRDs we can have multiple policies with the same name - across namespaces.